### ISSUE TITLE
Documentation lacks v8 model content and switching between versions is impossible

### ISSUE THREAD

**Issue by zmuxuny** [ultralytics/ultralytics#21804]
I want to give some honest feedback about your documentation. Almost everything is about the v11 model, but I just want to use the v8 model. There is no way to find usage instructions or training methods for v8. The documentation experience is extremely frustrating and unhelpful.

I sincerely hope you can provide a way to switch between documentation for different versions (especially v8 and v11). Please add and improve the documentation for v8, so users can easily learn and use it. Right now, it's nearly impossible for anyone who wants to work with v8.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @zmuxuny, thank you for your thoughtful feedback on our documentation experience 🚀. This is an automated response to help get you unblocked quickly—an Ultralytics engineer will also assist you here soon.

We understand your point about wanting clearer paths for YOLOv8 content and version switching. While our Docs highlight the latest YOLO11, we want to make sure YOLOv8 users can still find what they need. To help us improve, could you please share:
- The specific topics you were looking for in YOLOv8 (e.g., training, validation, export, deployment)?
- Any pages you checked that felt incomplete or confusing?
- Your target environment and workflow (CLI or Python)?

If this is a 🐛 Bug Report about missing/incorrect documentation content, please provide a minimum reproducible example of the issue you’re facing (e.g., exact commands or code snippets that don’t match doc expectations) to help us diagnose it faster:
- [Minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)

In the meantime, many fundamentals (datasets, training commands, Python/CLI usage) are similar across versions. You may find these sections helpful:
- [Docs](https://docs.ultralytics.com/)
- [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples
- [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Bovey0809** [comment]
@zmuxuny Which page are you referring to? I think you can simply change 11 to v8 for any training command.

**Comment by Y-T-G** [comment]
Although, most of the example code for the tasks will show YOLO11 as example, you can simply swap the model name to use YOLOv8 like @Bovey0809 mentioned.

Also example code is also provided in the YOLOv8 page:
https://docs.ultralytics.com/models/yolov8/#__tabbed_2_1

**Comment by glenn-jocher** [comment]
@zmuxuny you load any YOLOv8 model easily with the latest `ultralytics` version:

```python
from ultralytics import YOLO

# Load a YOLOv8 model
model = YOLO("yolov8n.pt")  # or any other size here

# Run inference
results = model("path/to/image.jpg")
```

**Comment by zmuxuny** [comment]
Hello, thank you all for the quick and helpful replies, especially @Bovey0809, @Y-T-G, and @glenn-jocher. I appreciate you taking the time to respond.

Let me provide more context for my feedback, which hopefully clarifies my concerns.

My main goal is to deploy a **YOLOv8 Pose model** on a resource-constrained device, the **OrangePi AI PRO**. I also plan to train a custom model for my work. I chose to start with v8 because I'm concerned that v11 might require more resources and seems to have less community validation for this kind of edge device so far.

Here were my specific points of confusion:

1.  **Finding Pre-trained Models:** My initial plan was to download a pre-trained `v8-pose` model to validate its performance on my device first, before attempting to train my own. However, I could only find direct download links for the v11 models on the documentation pages. This is what triggered my initial frustration.

2.  **Development Workflow & Codebase:** My biggest confusion comes from the unified repository. Since I plan to fine-tune a pre-trained model and might need to modify some code for my project's specific needs, I was looking for a v8-specific codebase or branch to clone. I wasn't sure what the best practice is. Should I just clone the main repository? If so, how can I be sure I am building upon the v8 architecture without accidentally pulling in dependencies or behaviors from v11?

In short, the consolidation of versions into a single repository is a bit confusing for a user who wants to "lock in" and develop on a specific older version. I was worried that subtle differences between v8 and v11 in training or exporting could cause issues down the line for my specific deployment target.

Thank you again for your help. Any guidance on the recommended workflow for custom training and development on v8 would be highly appreciated.

**Comment by Y-T-G** [comment]
Ultralytics is a library that provides the ability to train multiple different models. It's not limited to a single model.

The pre-trained models and their download links are provided in the respective docs pages of the models.

The page for YOLOv8 page, as linked previously, is here:
https://docs.ultralytics.com/models/yolov8/#performance-metrics

> Since I plan to fine-tune a pre-trained model and might need to modify some code for my project's specific needs, I was looking for a v8-specific codebase or branch to clone. I wasn't sure what the best practice is. Should I just clone the main repository? If so, how can I be sure I am building upon the v8 architecture without accidentally pulling in dependencies or behaviors from v11?

There is no YOLOv8 specific codebase. Ultralytics is designed to support multiple models and architectures. The architectures are defined in YAML files which you can find in the `cfg` folder:
https://github.com/ultralytics/ultralytics/tree/main/ultralytics/cfg/models/v8

You can clone the main repository and make the changes as necessary.

>  I was worried that subtle differences between v8 and v11 in training or exporting could cause issues down the line for my specific deployment target.

There shouldn't be any differences in training or exporting between YOLOv8 and YOLO11. Ultralytics is designed to be easy to use and you only need to swap the model filenames to switch between YOLOv8 and YOLO11.

For the Docs, I would suggest using the Ask AI option on the page if you're not able to find the relevant page. It should provide you with the relevant pages. And you can also specifically for things like pretrained model links.

**Comment by zmuxuny** [comment]
Hi @Y-T-G,

Thank you so much for the detailed and clear explanation!

Your response perfectly clarifies the philosophy behind the unified repository. Understanding that Ultralytics is a library designed to support multiple models, with architectures defined in the YAML files, makes complete sense now. This was the key piece of information I was missing.

The direct link to the v8 performance metrics and the confirmation that the training and exporting workflows are consistent across versions are also incredibly helpful. This resolves all my confusion.

I really appreciate you taking the time to break it down for me. I'll go ahead and close this issue now.

Thanks again to everyone who helped!
